src/App.jsx

```diff
@@ -1,33 +1,63 @@
+import { useState } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+function filterMovie({ title, description }, query) {
+  const queryLower = query.toLowerCase();
+  const titleLower = title.toLowerCase();
+  const descriptionLower = description.toLowerCase();
+
+  return titleLower.includes(queryLower)
+    || descriptionLower.includes(queryLower);
+}
+
+function getPrepareMovies(movies, query) {
+  let preparedMovies = [...movies];
+
+  if (query) {
+    preparedMovies = preparedMovies
+      .filter(movie => filterMovie(movie, query));
+  }
+
+  return preparedMovies;
+}
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const visibleMovies = getPrepareMovies(moviesFromServer, query);
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={(event) => {
+                  setQuery(event.target.value.trim());
+                }}
```

Comment:
I think `.trim()` it is a part of `getPrepareMovies` logic
```suggestion
                onChange={(event) => setQuery(event.target.value)}
```